src/Sylius/Bundle/AdminBundle/Action/RemoveAvatarAction.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Action;
+
+use Sylius\Component\Core\Model\AvatarImage;
+use Sylius\Component\Core\Repository\AvatarImageRepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+
+final class RemoveAvatarAction
+{
+    /** @var AvatarImageRepositoryInterface */
+    private $avatarRepository;
+
+    /** @var EngineInterface */
+    private $templatingEngine;
+
+    /** @var RouterInterface */
+    private $router;
+
+    public function __construct(
+        AvatarImageRepositoryInterface $avatarRepository,
+        EngineInterface $templatingEngine,
+        RouterInterface $router
+    ) {
+        $this->avatarRepository = $avatarRepository;
+        $this->templatingEngine = $templatingEngine;
+        $this->router = $router;
+    }
+
+    public function __invoke(Request $request): Response
```

Comment:
This action should be protected by the csrf token. We need to render csrf in the template, send it together with form and assert if the token is valid